src/coreclr/gc/gc.cpp

```diff
@@ -11654,6 +11656,36 @@ void gc_heap::clear_gen0_bricks()
     }
 }
 
+void gc_heap::check_gen0_bricks()
+{
+//#ifdef _DEBUG
```

Comment:
what are the perf implications of this check?

Reply:
it's not expensive but we still wouldn't want to actually ship this in retail. I think what Peter was suggesting was we'd keep this for now and get rid of it when we are confident that we've found all the race conditions (which certainly should be before we actually ship the next version).

Reply:
Exactly what Maoni said - let's leave it in for now, remove in a couple months after we have been running with regions as the default for a while.